analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -169,6 +193,62 @@ class Pub(
 
     override fun getVersionRequirement(): Requirement = Requirement.buildIvy("[2.2,)")
 
+    // Bootstrap flutter if it's not yet installed
+    override fun beforeResolution(definitionFiles: List<File>) {
+        if (getPathFromEnvironment(flutterCommand) != null) {
+            log.info { "Skipping to bootstrap flutter as it is already in PATH." }
+            return
+        }
+
+        if (flutterAbsolutePath.resolve(flutterCommand).isFile) {
+            log.info { "Skipping to bootstrap flutter as it was found in $flutterAbsolutePath." }
+            return
+        }
+
+        log.info { "Bootstrapping flutter as it was not found in PATH nor bootstrapped before." }
```

Comment:
Nit: Strictly speaking, besides being in PATH or bootstrapped, flutter might also have been found via `Os.env["FLUTTER_HOME"]`. So how about not trying to be overly specific and just say "... as it was not found."